text/0000-floating-point-additional-precision.md

```diff
@@ -0,0 +1,141 @@
+- Feature Name: `allow_extra_fp_precision`
+- Start Date: 2019-04-08
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Update the Rust specification to allow floating-point operations to provide
+*more* precision than specified, but not less precision; this allows many safe
+optimizations.
+
+# Motivation
+[motivation]: #motivation
+
+Some platforms provide instructions to run a series of floating-point
+operations quickly, such as fused multiply-add instructions; using these
+instructions can provide performance wins up to 2x or more. These instructions
+may provide *more* precision than required by IEEE floating-point operations,
+such as by doing multiple operations before rounding or losing precision.
+Similarly, high-performance floating-point code could perform multiple
+operations with higher-precision floating-point registers before converting
+back to a lower-precision format.
+
+In general, providing more precision than required should not cause a
+mathematical algorithm to fail or to lose numeric accuracy.
+
+This RFC proposes allowing floating-point types to perform intermediate
+calculations using more precision than the type itself, as long as they provide
+*at least* as much precision as the IEEE 754 standard requires.
+
+See the [prior art section](#prior-art) for precedent in several other
+languages and compilers.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+After this RFC, we could explain floating-point operations as follows:
+
+Floating-point operations in Rust have a guaranteed minimum accuracy, which
+specifies how far the result may differ from an infinitely accurate,
+mathematically exact answer. The implementation of Rust for any target platform
+must provide at least that much accuracy. In some cases, Rust can perform
+operations with higher accuracy than required, and doing so provides greater
+performance (such as by removing intermediate rounding steps). This will never
+make any floating-point operation *less* accurate, but it can make
+floating-point operations *more* accurate, making the result closer to the
+mathematically exact answer.
+
```

Comment:
This does not mention that floating-point arithmetic results can change depending on the optimization level (and potentially other compiler options?), toolchain version (new toolchain performs a new optimization by default that changes results), etc.